### PR TITLE
WebUI: edituser page with new Auth system

### PR DESCRIPTION
**Pull request by PipoCanaja**
Hi team,

This PR was started because I lacked the "Edit Permission" capability for Active Directory users. I tried to find some kind of "minimal" changeset to achieve this.

And I like the little encouragement cause **Edit Permissions** works so far :) (tested it 5 minutes, time needed to create and edit permission on one new user) 

I'll continue digging in this direction, but I would like to get your opinion and advice (and probably help as well) on this. I suppose you would prefer the complete refactor of the page (Auth object, Device object, getting rid of all select/update DB, etc etc), but I don't feel comfortable at all starting this, so I started "step by step" for the moment. 

Let me know,

Thx,

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Definitely something that is needed.  You might as well refactor it to a Laravel page.  Put an route in web.php, create a controller and make a blade template.

**Comment by PipoCanaja** [comment]
That's more or less the answer I expected ;) But I have absolutely no clue on how to do that :) I'll check if I get an idea on how to do it and depending on that, we'll decide if I go for it or not.

**Comment by murrant** [comment]
Places to start :)

https://laracasts.com/series/laravel-from-scratch-2017/episodes/2
https://laracasts.com/series/laravel-from-scratch-2017/episodes/5
https://laracasts.com/series/laravel-from-scratch-2017/episodes/8


https://laravel.com/docs/5.4/routing
https://laravel.com/docs/5.4/controllers
https://laravel.com/docs/5.4/blade

**Comment by PipoCanaja** [comment]
Hello

I'll remove the WIP for the moment and let this PR be merged. I'll try to get the Laravel logik and do some experiments "off" but I need this PR into master so I can use it to create my users ;) 

Thanx

File: html/pages/edituser.inc.php [hunk 1]
```diff
@@ -10,46 +11,49 @@
...
 $pagetitle[] = 'Edit user';
 
-if (!LegacyAuth::user()->isAdmin()) {
+if (! Auth::user()->hasGlobalAdmin()) {
     include 'includes/error-no-perm.inc.php';
 } else {
     if ($vars['user_id'] && !$vars['edit']) {
-        $user_data = LegacyAuth::get()->getUser($vars['user_id']);
+        /** @var User $user */
+        $user = User::all()->where('user_id', $vars['user_id'])->first();
```
Comment (murrant) [inline html/pages/edituser.inc.php]: FYI, this is the long way of writing `User::find($vars['user_id']);`

File: html/pages/edituser.inc.php [hunk 2]
```diff
@@ -432,7 +436,7 @@
             }//end if !empty($users_details)
         }//end if
     } else {
-        $user_list = LegacyAuth::get()->getUserlist();
+        $userlist = User::all();
```
Comment (murrant) [inline html/pages/edituser.inc.php]: You need to use `User::thisAuth()->all();` to exclude users from inactive auths.

**Comment by murrant** [comment]
@PipoCanaja can you fix the queries I've pointed out and we'll merge this now and rewrite later :)

File: html/pages/edituser.inc.php [hunk 3]
```diff
@@ -264,7 +268,7 @@
                 }
             }
 
-            $users_details = LegacyAuth::get()->getUser($vars['user_id']);
+            $users_details = User::all()->where('user_id', $vars['user_id'])->first()->toArray();
```
Comment (Jellyfrog) [inline html/pages/edituser.inc.php]:
```suggestion
            $users_details = User::find($vars['user_id'])->toArray();
```

**Pull request merged**